oarepo_ui/theme/assets/semantic-ui/js/oarepo_ui/forms/hooks.js

```diff
@@ -167,6 +167,7 @@ export const useDepositApiClient = (
   const [isSaving, setIsSaving] = useState(false);
   const [isPublishing, setIsPublishing] = useState(false);
   const [isDeleting, setIsDeleting] = useState(false);
+  const [isPreviewing, setIsPreviewing] = useState(false);
```

Comment:
There becomes way too many state variables, just to disable buttons, where there's some activity on the form. I would suggest merging all into a single state, like `isProcessing` ?

Reply:
Ahoj, initially, it was all just the isSubmitting variable from formik, but the problem with that is that I would give the loading spinner to all buttons i.e. i click on save, publish, delete, preview would get a spinner, which i found to be ugly and also potentially missleading in case the request would hang for a while. If you wish, i can put it back that way.

Reply:
I think spinners on all actions, or some spinner overlay over the whole buttons section is okay. Users shouldn't be able to click e.g. on publish, when there's some other action (like delete) already ongoing

Reply:
@mirekys Ahoj, though currently not implemented well, the idea was to use isSubmitting, to disable all the buttons via disabled prop, but for specific boolean to control if the button is loading or not. So for example, for save button, loading = {isSaving}, disabled={isSubmitting}. Anyway, if we wish to use just one boolean for all buttons, then I propose I simply use the formik's isSubmitting boolean for that, and I will remove all the other state variables (isSaving, isPublishing and so on). Just a show of what it looks like in that case

![image](https://github.com/oarepo/oarepo-ui/assets/34883098/59eed03b-44eb-4512-9f86-662aa41e6a32)

Please let me know and thanks!

Reply:
@mirekys Ahoj, I've made the changes as discussed, please take a look. Note: related PR https://github.com/Narodni-repozitar/nr-docs/pull/102 and publishbuttoncomponent will be removed from nr-model, leaving only the one in oarepo ui, which I think makes more sense overall.